Microsoft.AspNetCore.Components.Server/Microsoft.AspNetCore.Components.Server.Circuits/RemoteRenderer.cs

```diff
@@ -0,0 +1,211 @@
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.Server.Circuits
+{
+	internal class RemoteRenderer : Renderer
+	{
+		internal readonly struct UnacknowledgedRenderBatch
+		{
+			public long BatchId
+			{
+				[CompilerGenerated]
+				get
+				{
+					throw null;
+				}
+			}
+
+			public ArrayBuilder<byte> Data
+			{
+				[CompilerGenerated]
+				get
+				{
+					throw null;
+				}
+			}
+
+			public TaskCompletionSource<object> CompletionSource
+			{
+				[CompilerGenerated]
+				get
+				{
+					throw null;
+				}
+			}
+
+			public Microsoft.Extensions.Internal.ValueStopwatch ValueStopwatch
+			{
+				[CompilerGenerated]
+				get
+				{
+					throw null;
+				}
+			}
+
+			public UnacknowledgedRenderBatch(long batchId, ArrayBuilder<byte> data, TaskCompletionSource<object> completionSource, Microsoft.Extensions.Internal.ValueStopwatch valueStopwatch)
+			{
+				throw null;
+			}
+		}
+
+		private static class Log
+		{
+			private static class EventIds
+			{
+				public static readonly EventId UnhandledExceptionRenderingComponent;
+
+				public static readonly EventId BeginUpdateDisplayAsync;
+
+				public static readonly EventId SkipUpdateDisplayAsync;
+
+				public static readonly EventId SendBatchDataFailed;
+
+				public static readonly EventId CompletingBatchWithError;
+
+				public static readonly EventId CompletingBatchWithoutError;
+
+				public static readonly EventId ReceivedDuplicateBatchAcknowledgement;
+
+				public static readonly EventId FullUnacknowledgedRenderBatchesQueue;
+			}
+
+			public static void SendBatchDataFailed(ILogger logger, Exception exception)
+			{
+				throw null;
+			}
+
+			public static void UnhandledExceptionRenderingComponent(ILogger logger, Exception exception)
+			{
+				throw null;
+			}
+
+			public static void BeginUpdateDisplayAsync(ILogger logger, string connectionId, long batchId, int dataLength)
+			{
+				throw null;
+			}
+
+			public static void BufferingRenderDisconnectedClient(ILogger logger, string connectionId)
+			{
+				throw null;
+			}
+
+			public static void CompletingBatchWithError(ILogger logger, long batchId, string errorMessage, TimeSpan elapsedTime)
+			{
+				throw null;
+			}
+
+			public static void CompletingBatchWithoutError(ILogger logger, long batchId, TimeSpan elapsedTime)
+			{
+				throw null;
+			}
+
+			public static void ReceivedDuplicateBatchAck(ILogger logger, long incomingBatchId)
+			{
+				throw null;
+			}
+
+			public static void FullUnacknowledgedRenderBatchesQueue(ILogger logger)
+			{
+				throw null;
+			}
+		}
+
+		internal readonly ConcurrentQueue<UnacknowledgedRenderBatch> _unacknowledgedRenderBatches;
+
+		public override Dispatcher Dispatcher
+		{
+			[CompilerGenerated]
+			get
+			{
+				throw null;
+			}
+		}
+
+		/// <summary>
+		/// Notifies when a rendering exception occurred.
+		/// </summary>
+		public event EventHandler<Exception> UnhandledException
```

Comment:
If we're treating UnhandledException as non-nullable here, should we remove the `UnhandledException?` checks that we have in the codebase?

Or set it as nullable if that is what we really mean...